vault/identity_store_entities.go

```diff
@@ -475,7 +490,79 @@ func (i *IdentityStore) pathEntityNameDelete() framework.OperationFunc {
 	}
 }
 
-func (i *IdentityStore) handleEntityDeleteCommon(ctx context.Context, txn *memdb.Txn, entity *identity.Entity) error {
+// pathEntityIDDelete deletes the entity for a given entity ID
+func (i *IdentityStore) handleEntityBatchDelete() framework.OperationFunc {
+	return func(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
+		entityIDs := d.Get("entity_ids").([]string)
+		if len(entityIDs) == 0 {
+			return logical.ErrorResponse("missing entity ids to delete"), nil
+		}
+
+		// Sort the ids by the bucket they will be deleted from
+		byBucket := make(map[string]map[string]struct{})
```

Comment:
I think here we can do away with `make(map[string][]string)`, and thereby avoid transforming the map back to slice before calling deleteIdsForBucket. No?

Reply:
I _think_ it may be better this way since we cannot know the number of entities going into each bucket. We'd have to `append()` which could get costly if we need to relocate often.